pipeline.go

```diff
@@ -263,7 +263,7 @@ func (p *PipelineDefinition) Pipelines() (*Pipelines, error) {
 		for name := range selectedSteps {
 			queue = append(queue, name)
 		}
-		ignoreNotSelected = len(queue) > 0
+		ignoreNotSelected := len(queue) > 0
```

Comment:
I don't see why this needs to be a variable and not just moved into the only `if` that it is used in. Also I'm still not sure that you really need to check for this there

Reply:
queue is empty after the for loop, thus I can't check if `len(queue) > 0` as it will always be 0, if queue is not empty the loop will not exit

Reply:
I think the solution could be to check in the if if `len(selectedSteps) > 0`